src/stream/src/from_proto/group_top_n_appendonly.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::hash::{HashKey, HashKeyDispatcher};
+use risingwave_common::types::DataType;
+use risingwave_common::util::sort_util::OrderPair;
+use risingwave_pb::stream_plan::GroupTopNNode;
+
+use super::*;
+use crate::common::table::state_table::StateTable;
+use crate::executor::{ActorContextRef, AppendOnlyGroupTopNExecutor};
+use crate::task::AtomicU64Ref;
+
+pub struct AppendOnlyGroupTopNExecutorBuilder;
+
+#[async_trait::async_trait]
+impl ExecutorBuilder for AppendOnlyGroupTopNExecutorBuilder {
```

Comment:
I'm considering whether it's better to also make `AppendOnly` a const generic. 🥵 So that we can reuse more codes, both in `from_proto` and `executor`.

Reply:
+1, but lazy 🥵 since copy-paste isn't that painful now